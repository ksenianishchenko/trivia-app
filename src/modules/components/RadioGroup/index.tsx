import React from "react";
import { TriviaQuestionItemAnswersChoice } from "../../../abstractions/api/models/triviaQuestionItem";

type RadioGroupTypes = {
  option: TriviaQuestionItemAnswersChoice;
  index: number;
}

const RadioGroup = (props: RadioGroupTypes) => {
  const {option, index} = props;

  return <div className="radio-group" key={index}>
    <label className="radio-group__label" htmlFor={option.id}>
        <input
            className="radio-group__input"
            type="radio"
            name={option.text}
            id={option.id}
        />
        <span>{option.text}</span>
    </label>
  </div>
}

export default RadioGroup;