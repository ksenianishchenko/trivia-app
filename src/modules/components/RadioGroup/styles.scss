.radio-group.error {
  color: crimson;
}

.radio-group.correct {
  color: darkgreen;
}