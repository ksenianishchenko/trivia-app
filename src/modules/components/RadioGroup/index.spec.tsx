import React from "react";
import { shallow } from 'enzyme';
import RadioGroup from "./index";


describe("should render <RadioGroup /> component correct", () => {

  const mockOption = {
    "id": "0",
    "text": "Lord Volan De Mort",
    "imageUrl": ""
  }

  it("should render radio group button", () => {
      const tree = shallow(<RadioGroup option={mockOption} index={1} />);
      expect(tree).toMatchSnapshot();
  })

})