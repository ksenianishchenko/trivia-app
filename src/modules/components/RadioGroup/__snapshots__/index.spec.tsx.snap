// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <RadioGroup /> component correct should render radio group button 1`] = `
<div
  className="radio-group"
>
  <label
    className="radio-group__label"
    htmlFor="0"
  >
    <input
      className="radio-group__input"
      id="0"
      name="Lord Volan De Mort"
      type="radio"
    />
    <span>
      Lord Volan De Mort
    </span>
  </label>
</div>
`;
